### PR TITLE
Closes #18 - Create basic markup for auth and setup sending data from forms to server

### LINKED ISSUE #18: Create basic markup for auth and setup sending data from forms to server

### PR DESCRIPTION
**Pull request by romanolytvyn**

### CODE REVIEW

**Review by OBats — APPROVED** [review]

File: src/api/http.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+const getToken = () => localStorage.getItem('access-token');
+
+const get = (url, params) => axios.get(url, {
+  params,
+  headers: {
+    'access-token': getToken(),
+  },
+});
+const post = (url, params) => {
+  console.log(url, params);
```
Comment (Antoniozinchenko) [inline src/api/http.js]: remove console.log

File: src/api/http.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  headers: {
+    'access-token': getToken(),
+  },
+});
+
+export default {
+  get,
+  post,
+  put,
+  del,
```
Comment (Antoniozinchenko) [inline src/api/http.js]: rename 
delete: del

**Review by rostyslav-vasyliuk — APPROVED** [review]

File: src/components/auth/SignIn.jsx [hunk 3]
```diff
@@ -1,6 +1,82 @@
...
+import Link from 'react-router-dom/Link';
+import { signIn } from '../../api/auth-api';
 
-const SignIn = () => <div>SignUp</div>;
+const SignIn = () => (
+  <div>
+    <Grid className="Auth" centered verticalAlign="middle">
+      <Grid.Column className="Form" width={8}>
+        <Segment>
+          <Formik
```
Comment (Antoniozinchenko) [inline src/components/auth/SignIn.jsx:27]: add validation schema

File: src/components/auth/SignIn.jsx [hunk 4]
```diff
@@ -1,6 +1,82 @@
...
+                  value={values.password}
+                />
+                {/* {errors.password && touched.password && errors.password} */}
+                <br />
+                <Button
+                  fluid
+                  size="large"
+                  color="black"
+                  type="submit"
+                  // disabled={isSubmitting}
```
Comment (Antoniozinchenko) [inline src/components/auth/SignIn.jsx]: button should be disabled if form is not valid (email and pass are required)

File: src/components/auth/SignUp.jsx [hunk 5]
```diff
@@ -1,6 +1,95 @@
...
+import { signUp } from '../../api/auth-api';
 
-const SignUp = () => <div>SignUp</div>;
+const SignUp = () => (
+  <div>
+    <Grid className="Auth" centered verticalAlign="middle">
+      <Grid.Column className="Form" width={8}>
+        <Segment>
+          <Formik
+            initialValues={{
```
Comment (Antoniozinchenko) [inline src/components/auth/SignUp.jsx:30]: add validation schema

**Review by Antoniozinchenko — CHANGES_REQUESTED** [review]

File: src/api/http.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+const post = (url, params) => {
+  console.log(url, params);
+  return axios.post(url, {
+    params,
+    headers: {
+      'access-token': getToken(),
+    },
+  });
+};
+const put = (url, params) => axios.put(url, {
```
Comment (Antoniozinchenko) [inline src/api/http.js]: axios.put(url, params, {headers})
headers options is 3rd arg

File: src/api/http.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+const get = (url, params) => axios.get(url, {
+  params,
+  headers: {
+    'access-token': getToken(),
+  },
+});
+const post = (url, params) => {
+  console.log(url, params);
+  return axios.post(url, {
```
Comment (Antoniozinchenko) [inline src/api/http.js]: axios.post(url, params, {headers})
headers options is 3rd arg

File: src/api/http.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    },
+  });
+};
+const put = (url, params) => axios.put(url, {
+  params,
+  headers: {
+    'access-token': getToken(),
+  },
+});
+const del = (url, params) => axios.delete(url, {
```
Comment (Antoniozinchenko) [inline src/api/http.js]: axios.delete(URL, params, header);

**Review by Antoniozinchenko — CHANGES_REQUESTED** [review]
https://github.com/axios/axios#request-method-aliases
change axios args

File: src/api/http.js (hunk 2 above)
Comment (romanolytvyn) [reply src/api/http.js]: I can't pick word "delete", it invoke an error (I think it is reserved word)

**Review by Roman6917 — APPROVED** [review]

File: src/api/http.js (hunk 2 above)
Comment (Antoniozinchenko) [reply src/api/http.js]: export default {
  get,
  post,
  put,
  delete: del
}

**Review by OBats — APPROVED** [review]

**Review by anzhelikavelychko — APPROVED** [review]

**Review by AndriyTykhan — APPROVED** [review]

**Pull request closed without merging**